lib/charms/grafana_agent/v0/cos_agent.py

```diff
@@ -361,6 +378,13 @@ def trigger_refresh(self, _):
         # FIXME: Figure out what we should do here
         self.on.data_changed.emit()
 
+    @staticmethod
+    def _relation_unit(relation: Relation) -> Optional[Unit]:
+        """Return the principal unit for a relation."""
+        if relation and relation.units:
+            return next(iter(relation.units))
```

Comment:
**question:** Is the implied order guaranteed? I.e. is the principal guaranteed to be the first item in the list?